src/ecdh.rs

```diff
@@ -184,6 +134,12 @@ impl SharedSecret {
     }
 }
 
+impl From<[u8; 32]> for SharedSecret {
+    fn from(inner: [u8; 32]) -> SharedSecret {
+        SharedSecret(inner)
+    }
+}
```

Comment:
It seems like this conversion should be the other way around.

Reply:
I don't understand why its the wrong way around? It converts from an array to a shared secret?

Reply:
Do you mean you'd like to see `From<SharedSecret> for [u8]`? I added `AsRef` impl as suggested above.

Reply:
Yes, I mean `From<SharedSecret> for [u8; 32]`

The problem with the opposite conversion is that it's almost implicit. Something like `foo(SharedSecret::from_raw_bytes(bytes))` is clearer than `foo(bytes.into())`.
The newtype is supposed to indicate "this is indeed a shared secret derived using ECDH, not some other 32 bytes coming from god-knows-where"

Reply:
Removed this function all together.